index.js

```diff
@@ -36,13 +78,32 @@ module.exports = function (opts = {}) {
     }
 
     app.authboot = {};
+
+    const concurrency = opts.maxAuthConcurrency
+      || options.maxAuthConcurrency
+      || app.config.get('auth:maxAuthConcurrency')
+      || 3;
+
+    const algorithm = opts.algorithm
+      || options.algorithm
+      || app.config.get('auth:algorithm')
+      || 'sha256';
+
+    let checker;
     const lookup = app.authboot.lookup = (lookupOpt || function ({ name, password }, callback) {
-      const hash = users.get(name);
-      if (!hash) {
+      const hashBuffer = users.get(name);
+      if (!hashBuffer) {
         debug(`unknown username ${name}`);
         return callback(null, false);
```

Comment:
do we care about the timings of guessing user names?  an invalid user name will return failure faster than an invalid password.

Reply:
this logic of failing faster on unknown username did not change in this PR and I think it unlikely to be of much use without the corresponding password, we could move this so it is in the coalesced calcuation though, I am neutral to this. These mitigations are for basic things and not meant to be complete for all scenarios

Reply:
Yep. completely unrelated to your change, just happened to notice it this time around :/